### PR TITLE
[Linux] Add JavaScript modal dialog

### PR DESCRIPTION
**Pull request by XiaosongWei**
Leverage components app_modal's JavaScript modal dialog implementation.

BUG=XWALK-5819

### CODE REVIEW

**Comment by XiaosongWei** [comment]
@pozdnyakov Please take a look, Thanks!

**Comment by rakuco** [comment]
+@darktears
wasn't something similar done for Windows recently? can't the code be shared?

**Comment by XiaosongWei** [comment]
It seems Windows could also leverage app_modal's implementation then we could achieve the same dialog look and feel as Chrome's.

**Comment by XiaosongWei** [comment]
@pozdnyakov @darktears @rakuco Can you please take a look at this? Deepin reported that quite a some of their Apps can't work as expected because of the suppressed javascript dialog and they want to migrate their apps from Chrome to Crosswalk asap. Thanks for you help!

File: runtime/browser/ui/xwalk_javascript_native_dialog_factory_views.cc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#if defined(USE_AURA)
+#include "ui/aura/window.h"
+#endif
+
+namespace {
+
+class XWalkJavaScriptNativeDialogViewsFactory
+    : public app_modal::JavaScriptNativeDialogFactory {
+ public:
+  XWalkJavaScriptNativeDialogViewsFactory() {}
```
Comment (pozdnyakov) [inline runtime/browser/ui/xwalk_javascript_native_dialog_factory_views.cc]: wouldn't '= default' work?

File: runtime/browser/ui/xwalk_javascript_native_dialog_factory_views.cc [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    dialog->web_contents()->GetDelegate()->ActivateContents(
+        dialog->web_contents());
+    gfx::NativeWindow parent_window =
+        dialog->web_contents()->GetTopLevelNativeWindow();
+#if defined(USE_AURA)
+    if (!parent_window->GetRootWindow()) {
+      // When we are part of a WebContents that isn't actually being displayed
+      // on the screen, we can't actually attach to it.
+      parent_window = NULL;
```
Comment (pozdnyakov) [inline runtime/browser/ui/xwalk_javascript_native_dialog_factory_views.cc]: nullptr

File: runtime/browser/ui/xwalk_javascript_native_dialog_factory_views.cc [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    return dialogViews;
+  }
+
+  DISALLOW_COPY_AND_ASSIGN(XWalkJavaScriptNativeDialogViewsFactory);
+};
+
+class XWalkConstrainedWindowViewsClient
+    : public constrained_window::ConstrainedWindowViewsClient {
+ public:
+  XWalkConstrainedWindowViewsClient() {}
```
Comment (pozdnyakov) [inline runtime/browser/ui/xwalk_javascript_native_dialog_factory_views.cc]: = default

File: runtime/browser/ui/xwalk_javascript_native_dialog_factory_views.cc [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+#include "ui/aura/window.h"
+#endif
+
+namespace {
+
+class XWalkJavaScriptNativeDialogViewsFactory
+    : public app_modal::JavaScriptNativeDialogFactory {
+ public:
+  XWalkJavaScriptNativeDialogViewsFactory() {}
+  ~XWalkJavaScriptNativeDialogViewsFactory() override {}
```
Comment (pozdnyakov) [inline runtime/browser/ui/xwalk_javascript_native_dialog_factory_views.cc]: ditto

**Comment by darktears** [comment]
Most probably I should have used app_modal in the first place on Windows :(. Didn't know it existed, content_shell doesn't even use that thing.

This PR should be reworked to include Windows as well and probably revert most of what I've landed in https://github.com/crosswalk-project/crosswalk/commit/b19a154ecbb1bd422b0489cac33c7aace9aae691.

Windows and Linux are similar, they use Aura, I don't why this PR wouldn't work on Windows.

**Comment by XiaosongWei** [comment]
@pozdnyakov Updated, Thanks!
@darktears I don't have window environment to verify this so I prefer to merge this patch firstly and then Windows team to submit another PR to enable app_modal's dialog on Windows.

**Comment by pozdnyakov** [comment]
lgtm

**Comment by pozdnyakov** [comment]
Implementation on Win is to be handled separately

**Comment by darktears** [comment]
@XiaosongWei in the future make sure at least your IT machine is setup for Crosswalk development or worst case install a VM. I spend time taking care of Android when I do work on cross platform code in Crosswalk, I could just throw it to the Android team and call it a day but no I try my best to help. Linux and Windows are very similar at content/ level, we need enabling on both OSes as they have equal high priority.

**Comment by darktears** [comment]
For the record, the code asserts in debug.

**Pull request merged**